src/main2.cpp

```diff
@@ -176,123 +122,38 @@ TexturedNormalledVertex texturedCubeVertexArray[] = {
-@param vertexIndices: The indices for the vertices of the sphere
-@param vertices: The vertices of the sphere
-@param normals: The normals of the sphere
-@UVs: The UVs of the sphere
-@return The IBO
-*/
-GLuint createVertexArrayElementObject2(std::vector<int> vertexIndices, std::vector<glm::vec3> vertices,
-	                                   std::vector<glm::vec3> normals, std::vector<glm::vec2> UVs)
-{
-	// Create a vertex array
-	GLuint vertexArrayObject;
-	glGenVertexArrays(1, &vertexArrayObject);
-	glBindVertexArray(vertexArrayObject);
-
-	glGenBuffers(1, &IBO);
-	glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, IBO);
-	glBufferData(GL_ELEMENT_ARRAY_BUFFER, vertexIndices.size() * sizeof(unsigned int), &vertexIndices.front(), GL_STATIC_DRAW);
-
-	// Upload Vertex Buffer to the GPU, keep a reference to it (vertexBufferObject)
-	GLuint vertexBufferObject;
-	glGenBuffers(1, &vertexBufferObject);
-	glBindBuffer(GL_ARRAY_BUFFER, vertexBufferObject);
-	//int check = arraySize;
-	glBufferData(GL_ARRAY_BUFFER, vertices.size() * sizeof(glm::vec3), &vertices.front(), GL_STATIC_DRAW);
-
-	glVertexAttribPointer(0,    // attribute 0 matches aPos in Vertex Shader
-		3,                      // size
-		GL_FLOAT,               // type
-		GL_FALSE,               // normalized?
-		1 * sizeof(glm::vec3),  // stride - each vertex contain 2 vec3 (position, color)
-		(void*)0                // array buffer offset
-	);
-	glEnableVertexAttribArray(0);
-
-	//Normals VBO setup taken from lab
-	GLuint normals_VBO;
-	glGenBuffers(1, &normals_VBO);
-	glBindBuffer(GL_ARRAY_BUFFER, normals_VBO);
-	glBufferData(GL_ARRAY_BUFFER, normals.size() * sizeof(glm::vec3), &normals.front(), GL_STATIC_DRAW);
-	glVertexAttribPointer(1, 3, GL_FLOAT, GL_FALSE, 1 * sizeof(glm::vec3), (GLvoid*)0);
-	glEnableVertexAttribArray(1);
-
-	//UVs VBO setup
-	GLuint uvs_VBO;
-	glGenBuffers(1, &uvs_VBO);
-	glBindBuffer(GL_ARRAY_BUFFER, uvs_VBO);
-	glBufferData(GL_ARRAY_BUFFER, UVs.size() * sizeof(glm::vec3), &UVs.front(), GL_STATIC_DRAW);
-	glVertexAttribPointer(2, 2, GL_FLOAT, GL_FALSE, 1 * sizeof(glm::vec3), (GLvoid*)0);
-	glEnableVertexAttribArray(2);
-
-	glBindBuffer(GL_ARRAY_BUFFER, 0);
-	glBindVertexArray(0);
-
-	return vertexArrayObject;
-}
-
 // Set aspect ratio of the window
 const int WIDTH = 1024, HEIGHT = 768;
 
-// Set/ declare some variables
-glm::vec3 eye(.0f, .350f*1.5, .7650f*1.5);
-glm::vec3 center(.00f, .0f, 0.0f);
-glm::vec3 up(0.0f, 1.0f, 0.0f);
-glm::vec3 GroupMatrixScale(1.0f, 1.0f, 1.0f);
-glm::mat4 groupMatrix;
-glm::mat4 rotationMatrixW = glm::rotate(glm::mat4(1.0f), glm::radians(0.0f), glm::vec3(1.0f, 1.0f, 1.0f));
-
+// State-related globals
+bool soundPlayed = false;
+bool scoreIncremented = false;
 int renderAs = GL_TRIANGLES;
 int shaderProgram;
 double lastMousePosX, lastMousePosY, lastMousePosZ;
 float FOV = 70;
 float AR = (float) WIDTH / (float) HEIGHT;
 float near = .01;
 float far = 50;
-float translateW = 0, translateY = 0, translateZ = 0;
+float translateW = 0;
```

Comment:
Im not sure these translations are used anymore

Reply:
(translateX,translateY,translateZ)

Reply:
I'll definitely look into it